lib/Support/OSCompatPosix.cpp

```diff
@@ -778,10 +778,12 @@ bool unset_env(const char *name) {
 void *SigAltStackLeakSuppressor::stackRoot_{nullptr};
 
 SigAltStackLeakSuppressor::~SigAltStackLeakSuppressor() {
+#if defined(__APPLE__) && !(defined(TARGET_OS_TV) && TARGET_OS_TV)
```

Comment:
My suggestion was to use just `defined(__APPLE__) && !TARGET_OS_TV`. As noted in my previous comment, doing this correctly also ensures that `TargetConditionals.h` is properly imported.

In addition to the suggested change above, we should also add a comment that this is not supported on tvOS.